ui/core-react/src/test/popup/Popup.test.tsx

```diff
@@ -322,653 +323,414 @@ describe("<Popup />", () => {
-      const wrapper = mount<Popup>(<Popup isOpen />);
-      const popup = wrapper.find(".core-popup").getDOMNode();
+      const spyOnClose = sinon.spy();
+      render(<Popup isOpen onClose={spyOnClose}/>);
+      const popup = screen.getByRole("dialog");
 
       const contextMenu = new MouseEvent("contextmenu");
       sinon.stub(contextMenu, "target").get(() => popup);
       window.dispatchEvent(contextMenu);
 
-      expect(wrapper.state().isOpen).true;
-      wrapper.unmount();
+      expect(spyOnClose).to.not.be.called;
+      expect(classesFromElement(screen.getByRole("dialog"))).to.not.include("core-popup-hidden");
     });
 
     it("should not hide when context menu used if pinned", () => {
-      const wrapper = mount<Popup>(<Popup isOpen isPinned />);
+      const spyOnClose = sinon.spy();
+      render(<Popup isOpen isPinned onClose={spyOnClose} />);
 
       const contextMenu = new MouseEvent("contextmenu");
       sinon.stub(contextMenu, "target").get(() => document.createElement("div"));
       window.dispatchEvent(contextMenu);
 
-      expect(wrapper.state().isOpen).true;
-      wrapper.unmount();
+      expect(spyOnClose).to.not.be.called;
+      expect(classesFromElement(screen.getByRole("dialog"))).to.not.include("core-popup-hidden");
     });
 
     it("should not hide when context menu used if closeOnContextMenu=false", () => {
-      const wrapper = mount<Popup>(<Popup isOpen closeOnContextMenu={false} />);
+      const spyOnClose = sinon.spy();
+      render(<Popup isOpen closeOnContextMenu={false} onClose={spyOnClose} />);
 
       const contextMenu = new MouseEvent("contextmenu");
       sinon.stub(contextMenu, "target").get(() => document.createElement("div"));
       window.dispatchEvent(contextMenu);
 
-      expect(wrapper.state().isOpen).true;
-      wrapper.unmount();
+      expect(spyOnClose).to.not.be.called;
+      expect(classesFromElement(screen.getByRole("dialog"))).to.not.include("core-popup-hidden");
     });
 
     it("should not hide when context menu used if onContextMenu prop is passed", () => {
+      const spyOnClose = sinon.spy();
       const spyContextMenu = sinon.spy();
-      const wrapper = mount<Popup>(<Popup isOpen onContextMenu={spyContextMenu} />);
+      render(<Popup isOpen onContextMenu={spyContextMenu} onClose={spyOnClose} />);
 
       const contextMenu = new MouseEvent("contextmenu");
       sinon.stub(contextMenu, "target").get(() => document.createElement("div"));
       window.dispatchEvent(contextMenu);
 
-      expect(wrapper.state().isOpen).true;
-      sinon.assert.called(spyContextMenu);
-      wrapper.unmount();
+      expect(spyOnClose).to.not.be.called;
+      expect(classesFromElement(screen.getByRole("dialog"))).to.not.include("core-popup-hidden");
+      expect(spyContextMenu).to.be.called;
     });
 
   });
 
   describe("keyboard handling", () => {
-    it("should close on Escape", () => {
+    it("should call onClose on Escape", async () => {
```

Comment:
Probably all of the test names should be changed, because the `isOpen` state is never actually read when rendering the component, only the props. So the open state is completely controlled, whenever a tests states that the popup "hides" or "closes", really all it did was call the "onClose" callback, it is on the user to actually close and update the "isOpen" prop. The component would need to be fixed, but that would be a breaking change...